From c7704692b4a6a8b281ee08f5fb67dfb5cf38489b Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Mon, 09 Dec 2024 14:29:03 +0200
Subject: [PATCH] [FSW-12160] fs1: pldm: remove ide

Remove IDE capability, as this feature will no longer be used.
PCIe capabilities is maintained using HW linked list, hence
removal of IDE capability is done by skipping IDE capability
in the list.

Change-Id: I744f56caa956a67cad52eb6f11be6f96bc8b88c8
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/arch/fs1/arc/palad.c b/arch/fs1/arc/palad.c
index 822bdf2..e4daf6c 100644
--- a/arch/fs1/arc/palad.c
+++ b/arch/fs1/arc/palad.c
@@ -158,6 +158,9 @@
 		val = num_vf << PCIE_DBI_TOTAL_VFS_INITIAL_VFS_REG_SRIOV_INITIAL_VFS_S |
 		      num_vf << PCIE_DBI_TOTAL_VFS_INITIAL_VFS_REG_SRIOV_TOTAL_VFS_S;
 		regmap_writel(val, IO0_H0_PCIE_DBI(mmPCIE_DBI_TOTAL_VFS_INITIAL_VFS_REG));
+		regmap_rmwritel(IO0_H0_PCIE_DBI(mmPCIE_DBI_FLIT_INJECT_CAP_HDR_REG),
+				mmPCIE_DBI_RESBAR_CAP_HDR_REG,
+				PCIE_DBI_FLIT_INJECT_CAP_HDR_REG_FNEXT_OFFSET_M);
 	}
 
 	/* skip ATS capability */
